### ISSUE TITLE
Migration to Null-Safety

### ISSUE THREAD

**Issue by deimantasa** [luanpotter/audioplayers#784]
Migrate existing code to `null-safety`.

**Full Description**
Since last week, Flutter 2.0 hit stable branch. With 2.0 we've received Dart 2.12 with new [sound null safety](https://dart.dev/null-safety) mechanism. Current `audioplayers` package doesn't support `null-safety` yet therefore migration is a must.

Scope - migration of both - `example` and `package` to null-safety.

**Example Code**
Migration will touch most of the code. Key objectives here is to utilise [dart migrate](https://dart.dev/null-safety/migration-guide) so all of the heavy lifting is done for us. And after that, run the build and fix places, that were fixed incorrectly by the tool.

**Platforms**
Migration will touch all of the platforms.

**Sample Files/URLs/Sources**
N/A.

**Mockups**
N/A.

**Research Resources**
`null-safety` is unavoidable thus sooner or later it must be done.
https://dart.dev/null-safety

**Additional**
This will be a heavy migration therefore careful code review is a must from one or more developers, after everything is done.

**Comment by RR-Reddy** [comment]
here is pull request #785

**Comment by Zeswen** [comment]
Waiting for this to update our application to Flutter 2.0!

It would be awesome to have it ASAP. Thank you for your great work :)

**Comment by hyungtaecf** [comment]
Can someone tell if it is already in progress?

I have some packages stuck on the migration because the audioplayers package depends on old versions of other packages...

**Comment by zenkog** [comment]
Hi while this is in progress, is there a way to override the dependencies in pubspec.yaml to include the null safety version ?
I need this ASAP

> here is pull request #785

**Comment by jannikbryld** [comment]
> Hi while this is in progress, is there a way to override the dependencies in pubspec.yaml to include this version directly from the github repository ?
> I need this ASAP
> 
> > here is pull request #785

```
audioplayers: 
    git: 
      url: https://github.com/wolfenrain/audioplayers.git
      ref: null-safety
```

Although what worked for me as a temporary solution, while awaiting the final release:

```
audioplayers: 0.17.4

dependency_overrides:
    uuid: ^3.0.1
    path_provider: ^2.0.1
```

**Comment by luanpotter** [comment]
Fixed by https://github.com/luanpotter/audioplayers/pull/785

**Comment by zenkog** [comment]
> Fixed by #785

Great work! 
Now how do I import the package? 
The `pub.dev` version is still the Feb 19th release.

**Comment by luanpotter** [comment]
I will release a new version later today!
